examples/resnet_50.py

```diff
@@ -0,0 +1,189 @@
+    """
+    the identity_block is the block that has no conv layer at shortcut
+    params:
+        input_tensor: input tensor
+        nb_filter: list of integers, the nb_filters of 3 conv layer at main path
+        stage: integer, current stage number
+        block: str like 'a','b'.., curretn block
+        kernel_size: defualt 3, the kernel size of middle conv layer at main path
+    """
+    nb_filter1, nb_filter2, nb_filter3 = nb_filter
+
+    out = Convolution2D(nb_filter1, 1, 1, name='res'+str(stage)+block+'_branch2a')(input_tensor)
+    out = BatchNormalization(axis=1, name='bn'+str(stage)+block+'_branch2a')(out)
+    out = Activation('relu')(out)
+
+    out = out = Convolution2D(nb_filter2, kernel_size, kernel_size, border_mode='same',
+                              name='res'+str(stage)+block+'_branch2b')(out)
+    out = BatchNormalization(axis=1, name='bn'+str(stage)+block+'_branch2b')(out)
+    out = Activation('relu')(out)
+
+    out = Convolution2D(nb_filter3, 1, 1, name='res'+str(stage)+block+'_branch2c')(out)
+    out = BatchNormalization(axis=1, name='bn'+str(stage)+block+'_branch2c')(out)
+
+    out = merge([out, input_tensor], mode='sum')
+    out = Activation('relu')(out)
+    return out
+
+
+def conv_block(input_tensor, nb_filter, stage, block, kernel_size=3):
+    """
+    conv_block is the block that has a conv layer at shortcut
+    params:
+        input_tensor: input tensor
+        nb_filter: list of integers, the nb_filters of 3 conv layer at main path
+        stage: integer, current stage number
+        block: str like 'a','b'.., current block
+        kernel_size: defualt 3, the kernel size of middle conv layer at main path
+
+    Note that from stage 3, the first conv layer at main path is with subsample=(2,2)
+    And the shortcut should has subsample=(2,2) as well
+    """
+    nb_filter1, nb_filter2, nb_filter3 = nb_filter
+    if stage == 2:
+        out = Convolution2D(nb_filter1, 1, 1, name='res'+str(stage)+block+'_branch2a')(input_tensor)
+    else:
+        out = Convolution2D(nb_filter1, 1, 1, subsample=(2, 2), name='res'+str(stage)+block+'_branch2a')(input_tensor)
+
+    out = BatchNormalization(axis=1, name='bn'+str(stage)+block+'_branch2a')(out)
+    out = Activation('relu')(out)
+
+    out = out = Convolution2D(nb_filter2, kernel_size, kernel_size, border_mode='same',
+                              name='res'+str(stage)+block+'_branch2b')(out)
+    out = BatchNormalization(axis=1, name='bn'+str(stage)+block+'_branch2b')(out)
+    out = Activation('relu')(out)
+
+    out = Convolution2D(nb_filter3, 1, 1, name='res'+str(stage)+block+'_branch2c')(out)
+    out = BatchNormalization(axis=1, name='bn'+str(stage)+block+'_branch2c')(out)
+
+    if stage == 2:
+        shortcut = Convolution2D(nb_filter3, 1, 1, name='res'+str(stage)+block+'_branch1')(input_tensor)
+    else:
+        shortcut = Convolution2D(nb_filter3, 1, 1, subsample=(2, 2), name='res'+str(stage)+block+'_branch1')(input_tensor)
```

Comment:
Instead pass an argument "stride" which will be used for subsampling here and above.